python/plugins/processing/algs/qgis/CheckValidity.py

```diff
@@ -0,0 +1,182 @@
+"""
+
+__author__ = 'Arnaud Morvan'
+__date__ = 'May 2015'
+__copyright__ = '(C) 2015, Arnaud Morvan'
+
+# This will get replaced with a git SHA1 when you do a git archive323
+
+__revision__ = '$Format:%H$'
+
+from PyQt4 import QtCore
+from qgis.core import QGis, QgsGeometry, QgsFeature, QgsField
+from processing.core.GeoAlgorithm import GeoAlgorithm
+from processing.core.parameters import ParameterVector
+from processing.core.parameters import ParameterSelection
+from processing.core.outputs import OutputVector
+from processing.tools import dataobjects, vector
+
+settings_method_key = "/qgis/digitizing/validate_geometries"
+
+
+class CheckValidity(GeoAlgorithm):
+
+    INPUT_LAYER = 'INPUT_LAYER'
+    METHOD = 'METHOD'
+    VALID_OUTPUT = 'VALID_OUTPUT'
+    INVALID_OUTPUT = 'INVALID_OUTPUT'
+    ERROR_OUTPUT = 'ERROR_OUTPUT'
+
+    METHODS = ['The one selected in digitizing settings',
+               'QGIS',
+               'GEOS']
+
+    def defineCharacteristics(self):
+        self.name = 'Check validity'
+        self.group = 'Vector geometry tools'
+
+        self.addParameter(ParameterVector(
+            self.INPUT_LAYER,
+            self.tr('Input layer'),
+            [ParameterVector.VECTOR_TYPE_ANY]))
+
+        self.addParameter(ParameterSelection(
+            self.METHOD,
+            self.tr('Method'),
+            self.METHODS))
+
+        self.addOutput(OutputVector(
+            self.VALID_OUTPUT,
+            self.tr('Valid output')))
+
+        self.addOutput(OutputVector(
+            self.INVALID_OUTPUT,
+            self.tr('Invalid output')))
+
+        self.addOutput(OutputVector(
+            self.ERROR_OUTPUT,
+            self.tr('Error output')))
+
+    def processAlgorithm(self, progress):
+        settings = QtCore.QSettings()
+        initial_method_setting = settings.value(settings_method_key, 1)
+
+        method = self.getParameterValue(self.METHOD)
+        if method != 0:
+            settings.setValue(settings_method_key, method)
+        try:
+            self.doCheck(progress)
+        except:
+            raise
+        finally:
+            settings.setValue(settings_method_key, initial_method_setting)
+
+    def doCheck(self, progress):
+        layer = dataobjects.getObjectFromUri(
+            self.getParameterValue(self.INPUT_LAYER))
+        provider = layer.dataProvider()
+
+        settings = QtCore.QSettings()
+        method = int(settings.value(settings_method_key, 1))
+
+        valid_ouput = self.getOutputFromName(self.VALID_OUTPUT)
+        valid_fields = layer.pendingFields().toList()
+        valid_writer = valid_ouput.getVectorWriter(
+            valid_fields,
+            provider.geometryType(),
+            layer.crs())
+        valid_count = 0
+
+        invalid_ouput = self.getOutputFromName(self.INVALID_OUTPUT)
+        invalid_fields = layer.pendingFields().toList() + [
+            QgsField(name='_validity_errors',
```

Comment:
Field name is too long for default shape output and will be truncated. Something shorter will be much better here